src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -6496,6 +6503,56 @@ private BoundExpression BindLeftIdentifierOfPotentialColorColorMemberAccess(Iden
             return boundValue;
         }
 
+        private BoundExpression RebindColorColorConstField(IdentifierNameSyntax left, BoundExpression boundValue, FieldSymbol leftSymbol)
+        {
+            if (!leftSymbol.IsConst)
+            {
+                return boundValue;
+            }
+
+            var memberAccessParent = left.Parent as MemberAccessExpressionSyntax;
+            Debug.Assert(memberAccessParent is not null);
```

Comment:
>Debug.Assert(memberAccessParent is not null); [](http://example.com/codeflow?start=12&length=45)

This assumption looks suspicious and probably is wrong.

Reply:
Do you think we can trigger this via a pointer access expression?